### ISSUE TITLE
[feature] Allow to use the profile name inside the cmake_layout build_folder_vars 

### ISSUE THREAD

**Issue by hpe-ykoehler** [conan-io/conan#13080]
### What is your suggestion?

Under this page https://docs.conan.io/en/latest/reference/conanfile/tools/cmake/cmake_layout.html#:~:text=For%20example%2C%20this%20would%20implement%20the%20standard%20CMake,and%20cpp%20attributes%20described%20in%20the%20%28layout%20reference%29.

One can adjust the build folder to match settings or options being used.  Yet in my case, we have a set of settings/options that we refer to as a "platform".  We know that the platform "ABC" is an arm, gcc 9.3, Linux, etc...

Now I can ask conan cmake_layout to create a build folder such as build/gcc/9.3/Linux/etc./Release, but really this is "decomposing" what my ABC platform actually means.

So I am looking at using the name of the profile itself as a variable to the then compute the build folder.

So that I could shortcut the above with build_folder_vars=["profile_host[0].name"] for example and obtain build/ABC/Release as the build folder out of cmake_layout.

I understand that there may be more than one profile, so using an array would be require and the end-user would know to pass the first or last profile as being the one that named the build folder.

Another way to achieve this could be to add a new settings.vendor.name or settings.platform or settings.name variable that could hold the string ABC in question.

### Have you read the CONTRIBUTING guide?

- [X] I've read the CONTRIBUTING guide

**Comment by perseoGI** [comment]
Any update on this?

**Comment by memsharded** [comment]
Quick question: what would be the name in the case?:
```
conan install -pr=basic -pr=augmentd -pr=special
```

**Comment by jputcu** [comment]
I'm in the same situation: trying to get a identifier in the build directory and also having a unique `cmake preset` name.
I started out by adding `build_folder_vars=["settings.arch"]`, but I have multiple variants with the same `arch`.
Could it be a solution to add a `options={"product":["ABC","BC","AC"]}` and then using `build_folder_vars=["options.product"]`?
Profile name would indeed also be an option.

**Comment by hpe-ykoehler** [comment]
@jputcu that is how I did it, you can add this to settings.yml, I used "platform" instead of "product"

```
   def layout(self):
        cmake_layout(self, build_folder=os.path.join("build", str(self.settings.os.platform)))

```

I added platform to settings "os" and then in my conan profile I make sure all my profile have a proper distincts value for the os.platform key.

**Comment by hpe-ykoehler** [comment]
I think this may even be a better solution for me than profile name, as one could call conan with more than one profile...

**Comment by memsharded** [comment]
https://github.com/conan-io/conan/pull/16633 introduced ``const.uservalue`` in the ``build_folder_vars`` specificacion, allowing users to use whatever name they want for their cmake layouts. If specified in the profile using the jinja variable ``profile_name`` it is possible to use it too. Closing this as solved in Conan 2.6.
